### ISSUE TITLE
Any chance of a Linux version?

### ISSUE THREAD

**Issue by EdyStauch** [aliasIsolation/aliasIsolation#27]
The title says it all. 

Thank you for the excellent work.

**Comment by foofthetoof** [comment]
I know this is an old post, but if anyone is wondering, the latest version of aliasIsolation (1.2.0) works on Linux. You just need to set proton compatibility mode in Steam. I used Proton 9.0-4.

To set proton compatibility:

- Right click _Alien: Isolation_ in your steam library
- _Properties_
- _Compatibility_

**Comment by 0CCULTIST** [comment]
> I know this is an old post, but if anyone is wondering, the latest version of aliasIsolation (1.2.0) works on Linux. You just need to set proton compatibility mode in Steam. I used Proton 9.0-4.
> 
> To set proton compatibility:
> 
> * Right click _Alien: Isolation_ in your steam library
> * _Properties_
> * _Compatibility_

Yes, but Alien Isolation has a Linux native version that would be preferable to use. 
We could use LD_PRELOAD to inject the libraries.

**Comment by foofthetoof** [comment]
> > I know this is an old post, but if anyone is wondering, the latest version of aliasIsolation (1.2.0) works on Linux. You just need to set proton compatibility mode in Steam. I used Proton 9.0-4.
> > To set proton compatibility:
> > 
> > * Right click _Alien: Isolation_ in your steam library
> > * _Properties_
> > * _Compatibility_
> 
> Yes, but Alien Isolation has a Linux native version that would be preferable to use. We could use LD_PRELOAD to inject the libraries.

It's been a long time since I have tried the Windows or Linux version now, but I remember having issues with the Linux version. Proton works so well now that I usually just prefer to run the Windows version.

**Comment by Deus-nsf** [comment]
I wanted to create a new blank issue to thank the contributors but I'll do it here.
I have it working on the Steam Deck on SteamOS even on top of RenoDX HDR with ReShade AND lossless scaling frame generation (45 x2 to 90 FPS), it's insane and works extremely well.
Again thanks so much for that! Don't forget to bind delete on a button to disable Alias sharpness and chroma if you're using RenoDX!

**Comment by RyanJGray** [comment]
> Yes, but Alien Isolation has a Linux native version that would be preferable to use. We could use LD_PRELOAD to inject the libraries.

This would be ideal and I would genuinely love to have a working version of Alias for the Linux native port that Feral did.

I'm going to get a bit technical as I want to write down why it hasn't really been explored yet, in case someone who knows more looks at this and decides to help out! 😄 

There's a couple of problems:
* Alias was built solely with Windows in mind, and it would take a fairly large amount of work to rework the code so that it works the same way on both Windows and Linux.
* Another complication is the way that Feral ported the game, they appear to have wrapped the Windows version of the game in a conversion layer called "InDirectX", which (I think) at compile-time converted the game's DirectX API calls to OpenGL equivalents. We would need to find out where the DirectX logic we were hooking into now lives in the Linux version, or the OpenGL equivalents.
* If I remember rightly, they also converted all the DirectX shaders to OpenGL via some custom conversion tool. We would need to rewrite all the shaders for the mod to work on top of OpenGL. We don't have the code for the shader that actually does the TAA resolve, and it would need to be rewritten or reverse-engineered.

It is possible, I just don't really know enough about OpenGL to do it right now.

With that said, I may make a start on at least splitting the current code up into Windows-specific and cross-platform sections to at least remove the first obstacle for the future.

**Comment by Deus-nsf** [comment]
Wait, I'm reading stuff about the Linux native version, but doesn't that version have issues over the Windows build though? It's only advantage is that it's a native 64 bit application, but other than that I believe there were issues with that version and that it doesn't have total parity with the Windows build no?

EDIT: Yes apparently the native Linux build have performance issues and other problems if I am to believe some posts on ProtonDB.

**Comment by RyanJGray** [comment]
@Deus-nsf Yeah, in my experience the native Linux build had serious stuttering issues. It is nice that it's 64-bit and all, but it wasn't as nice to play for me. I basically always use the Windows build under Proton.

64-bit builds of the game have a couple of extra features in terms of behaviour during gameplay - more monitors in game show animated content instead of just being frozen, lights on transit cars and lifts move instead of floating in place, and probably something else that I've forgotten. The reason PC was a 32-bit "old gen" (as CA called it) build is not really known, even to the former developers.

I believe @MattFiler has worked around all of that with OpenCAGE.

**Comment by Deus-nsf** [comment]
PC was probably 32 bit because the development of the original version was a cross gen game that had to work on last gen 32 bit consoles as well (xbox 360 / PS3) so they might have just carried that over for some reason? Or maybe it's just compatibility for the time who knows.

**Comment by RyanJGray** [comment]
Possibly, I remember a developer saying it was likely compatibility (not wanting to limit their market on PC).

Supporting the Linux port was never really high on my priority list as it didn't seem many people used it, same with macOS.

The highest thing on my priority list for Alias is controller UI support and rewriting the shaders (as we lack the code for the TAA shader).

The Windows version runs basically everywhere, thankfully! 😅

**Comment by Deus-nsf** [comment]
Yeah Alias UI is fine when using steam input, at least for devices with trackpads that is.
And yeah I'd say that the Linux native version can just be ignored, as Proton is set to prioritize the Windows build anyway.
It's great that we have TAA inputs thanks to this mod, that theoretically makes it possible to have FSR 4 running instead if some crazy people feel up to the task, or maybe as an Optiscaler plugin ^^
